c++/src/kj/compat/http.c++

```diff
@@ -6906,19 +6904,27 @@ private:
     // We only provide the Response object if we know we haven't already sent a response.
     auto promise = server.settings.errorHandler.orDefault(*this).handleApplicationError(
         kj::mv(exception), currentMethod.map([this](auto&&) -> Response& { return *this; }));
-
-    return promise.then([this]() { return httpOutput.flush(); })
-                  .then([]() { return false; });  // loop ends after flush
+    return finishSendingError(kj::mv(promise));
   }
 
   kj::Promise<bool> sendError() {
     closeAfterSend = true;
 
     // We can provide a Response object, since none has already been sent.
     auto promise = server.settings.errorHandler.orDefault(*this).handleNoResponse(*this);
+    return finishSendingError(kj::mv(promise));
+  }
 
-    return promise.then([this]() { return httpOutput.flush(); })
-                  .then([]() { return false; });  // loop ends after flush
+  kj::Promise<bool> finishSendingError(kj::Promise<void> promise) {
+    return promise.then([this]() -> kj::Promise<void> {
+      if (httpOutput.isBroken()) {
+        // Skip flush for broken streams, since it will throw an exception that may be worse than
+        // the one we just handled.
+        return kj::READY_NOW;
+      } else {
+        return httpOutput.flush();
+      }
```

Comment:
I'm not sure if this is necessarily the correct solution -- there are a lot of other ways to suppress the error, and doing it this way may hide stream errors originating in the handlers themselves.

But I do notice that there a couple other places in `HttpServer::Connection` that are also using the pattern of skipping the `flush()` call if `isBroken()` is true, so I assume it's at least a reasonable thing to do in particular situations.

Reply:
This seem pretty reasonable to me. I think this is a place where there isn't any perfect answer.